submissions/smillims/memory-pair-game/js/script.js

```diff
@@ -1,89 +1,110 @@
 'use strict';
 
-const allCards = document.querySelectorAll('.flipper');
+const cards = document.querySelector('.cards');
 
-const cardPosition = 12;
-const allPairCards = 6;
-let currentPair = 0;
-let hasFlippedCard = false;
 let lockCard = false;
 let firstCard = '';
 let secondCard = '';
 
-function flipCards() {
-	if(lockCard){
-		return;
-	};
-	if (this === firstCard){
+const images = [
+	'bentley',
+	'amazon',
+	'google',
+	'intel',
+	'nike',
+	'telegram'
+];
+
+const allImg = [...images, ...images];
+
+const numberOfallImg = allImg.length;
+const allPairInGame = 6;
+let currentPair = 0;
+
+function sortCard() {
+	allImg.sort(function() { 
+		return 0.5 - Math.random() 
+	});
+};
+
+function generateCard() {
+	sortCard();
+
+	let divFlipper = '';
+	allImg.forEach(item => 
+		divFlipper += `<div class="flipper" data-id="${item}">
+		<div class="front"><img src="img/logo.png" alt="Logo"></div>
+		<div class="back"><img src="img/${item}.png" alt="${item}"></div>
+	</div>`
+	);
+	
+	cards.innerHTML = divFlipper;
+	
+};
+
+function flipCards(e) {
+	let targetCard = e.target.closest('.flipper');
+
+	if (!targetCard) return;
+	if(lockCard || targetCard === firstCard) {
 		return;
 	};
 
-	this.classList.add('flip');
-
-	if(!hasFlippedCard) {
-		hasFlippedCard = true;
-		firstCard = this;
+	if(!firstCard) {
+		firstCard = targetCard;
+		firstCard.classList.add('flip');
 		return;
 	} else {
-		secondCard = this;
+		secondCard = targetCard;
+		secondCard.classList.add('flip');
 		checkPair();
 	}
 };
 
 function checkPair() {
-	if(firstCard.dataset.id === secondCard.dataset.id){
+	if(firstCard.dataset.id === secondCard.dataset.id) {
 		disablePair();
 		checkWin();
 	} else {
 		frontFlipPair();
 	}
-};
+}
 
 function disablePair() {
-	firstCard.classList.add('opacity');
 	firstCard.removeEventListener('click', flipCards);
-	secondCard.classList.add('opacity');
 	secondCard.removeEventListener('click', flipCards);
+	firstCard.classList.add('opacity');
+	secondCard.classList.add('opacity');
 
 	currentPair++;
 
 	resetValue();
 };
 
 function frontFlipPair() {
+	let waitingTime = 600;
```

Comment:
1. frontFlipPair will create variable `waitingTime` for every time when it execute
2. why `let` ?

Reply:
![изображение](https://user-images.githubusercontent.com/66283273/107973517-f7185080-6fbd-11eb-8cf0-9ff77f24d709.png)
how to declare it then?

> 2. why `let`?

because it could change